src/cmocka/rnp_tests_support.c

```diff
@@ -241,50 +249,84 @@ setupPassphrasefd(int *pipefd)
     }
 
     /*Write and close fd*/
     const char *password = "passwordforkeygeneration\0";
     assert_int_equal(write(pipefd[1], password, strlen(password)), strlen(password));
     close(pipefd[1]);
-    return 1;
+    return RNP_OK;
 }
 
 int
-setup_rnp_common(rnp_t *rnp, char *passfd)
+setup_rnp_common(rnp_t *                 rnp,
+                 enum key_store_format_t ks_format,
+                 const char *            homedir,
+                 int *                   pipefd)
 {
-    int rc;
-    /*Initialize the basic RNP structure. */
-    memset(rnp, '\0', sizeof(*rnp));
-    /*Set the default parameters*/
-    rc = rnp_setvar(rnp, "sshkeydir", "/etc/ssh");
-    if (rc != 1) {
-        return rc;
+    int          res;
+    char         pubpath[1024];
+    char         secpath[1024];
+    char         homepath[1024];
+    rnp_params_t params;
+
+    rnp_params_init(&params);
+
+    /* set password fd if any */
+    if (pipefd) {
+        if ((res = setupPassphrasefd(pipefd)) != 1) {
+            return res;
+        }
+        params.passfd = pipefd[0];
     }
-    rc = rnp_setvar(rnp, "res", "<stdout>");
-    if (rc != 1) {
-        return rc;
+    /* setup keyring pathes */
+    if (homedir == NULL) {
+        /* if we use default homedir then we append '.rnp' and create directory as well */
+        homedir = getenv("HOME");
+        paths_concat(homepath, sizeof(homepath), homedir, ".rnp", NULL);
+        if (!dir_exists(homepath))
+            path_mkdir(0700, homepath, NULL);
+        homedir = homepath;
     }
-    rc = rnp_setvar(rnp, "format", "human");
-    if (rc != 1) {
-        return rc;
+
+    if (homedir == NULL) {
+        return RNP_FAIL;
     }
-    rc = rnp_setvar(rnp, "userid", getenv("LOGNAME"));
-    if (rc != 1) {
-        return rc;
+
+    if ((ks_format != GPG_KEY_STORE) && (ks_format != KBX_KEY_STORE)) {
+        return RNP_FAIL;
     }
 
-    if (passfd) {
-        rc = rnp_setvar(rnp, "pass-fd", passfd);
-        if (rc != 1) {
-            return rc;
-        }
-        rc = rnp_setvar(rnp, "need seckey", "true");
-        if (rc != 1) {
-            return rc;
-        }
+    paths_concat(pubpath,
+                 sizeof(pubpath),
+                 homedir,
+                 (ks_format == GPG_KEY_STORE) ? "pubring.gpg" : "pubring.kbx",
+                 NULL);
+    paths_concat(secpath,
+                 sizeof(secpath),
+                 homedir,
+                 (ks_format == GPG_KEY_STORE) ? "secring.gpg" : "secring.kbx",
+                 NULL);
+    params.pubpath = strdup(pubpath);
+    params.secpath = strdup(secpath);
+    params.ks_format = ks_format;
+
+    /*initialize the basic RNP structure. */
+    memset(rnp, '\0', sizeof(*rnp));
+    if ((res = rnp_init(rnp, &params)) != 1) {
+        return res;
     }
+    rnp_params_free(&params);
 
-    return rnp_init(rnp);
+    return RNP_OK;
+}
+
+void
+set_default_rsa_key_desc(rnp_keygen_desc_t *key_desc, pgp_hash_alg_t hashalg)
```

Comment:
The same code was used in rnp_tests_exportkey and rnp_tests_generatekey, so I moved it here to reuse the code. Also 1024 bits were hardcoded, so I reused them.